### DISCUSSION TITLE
Installing Zigbee4Domoticz on OpenWRT

### DISCUSSION THREAD

**Discussion by pipiche38** [zigpy/zigpy#931] [Q&A]
I have a user whom is struggling to get Zigbee4Domoticz plugin running on OpenWRT platform

https://www.domoticz.com/forum/viewtopic.php?t=38117&sid=78774a9d76aa1a166f36dd6d07a77bfd

Is there any way to tweak teh library in order to reduce the dependencies +?

**Comment by puddly** [comment]
> pycrypto can't be installed to OpenWRT due to required system packages missing and unavailable.
> 
> For other dependencies listed, pip actually runs out of RAM (120Mb total) before succeeding. I will need to find a way to load these libraries.

This is a pretty esoteric build environment and probably not one that can be supported.

Some options:

1. Don't compile the Python packages on the embedded device, cross compile them and distribute the built Wheels: `pip install ./your-compiled-pycrypto.whl`
2. Create `site-packages/Crypto/Cipher/AES.py` to make a "fake" pycrypto if you don't plan on using install codes.  It will satisfy imports but break the moment you use `zigpy.util.convert_install_code`.
3. See if `pip install cryptography` works. If it does, we can switch from pycrypto to cryptography if you submit a PR.
4. Setup a temporary swap disk.

**Comment by innovate-invent** [comment]
Solution: https://github.com/zigbeefordomoticz/wiki/blob/zigpy/en-eng/Plugin_Installation.md#8---installation-running-on-openwrt
